src/Database/Behaviors/Purgeable.php

```diff
@@ -0,0 +1,125 @@
+<?php namespace October\Rain\Database\Behaviors;
+
+use Exception;
+
+class Purgeable extends \October\Rain\Extension\ExtensionBase
+{
+    /**
+     * @var array List of attribute names which should not be saved to the database.
+     *
+     * protected $purgeable = [];
+     */
+
+    protected $parent;
+
+    public function __construct($parent)
+    {
+        $this->parent = $parent;
+        $this->bootPurgeable();
+    }
+
+    /**
+     * @var array List of original attribute values before they were purged.
+     */
+    protected $originalPurgeableValues = [];
+
+    /**
+     * Boot the purgeable trait for a model.
+     * @return void
+     */
+    public function bootPurgeable()
+    {
+        if (!$this->parent->propertyExists('purgeable'))
```

Comment:
Just had an idea here, could you throw an exception if they don't include the `'purgeable'` entry in their `purgeable` property?

Reply:
Sure, do you know on the top of your head why it tries to save it to the db? When just a property dynamically it threw no sql errors but as soon as the behavior is implemented it would.

Reply:
So you're saying that if you just `addDynamicProperty` it doesn't cause any issues but if you add the behavior itself as well it does cause issues?

Reply:
Yup

Reply:
Ah, @LukeTowers When defining the $purgeable property it can't be declared protected like in can for the trait, the behavior has no idea it's acting on behalf of a class.

Reply:
I don't know a good reason why we would not be able to use a public var for this, even though I dont *like* it, it seems to be in line with how dynamic properties are declared?

Reply:
Okay so implementing the behavior directly in a class seems to work when $purgeable is public, extending the class throws [here](https://github.com/octobercms/library/blob/0c45460e23f5b9916f4eecf4d8d3382d5cd03e15/src/Database/Model.php#L519)